src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -219,6 +225,7 @@ public SlaveTemplate(String ami, String zone, SpotConfiguration spotConfig, Stri
         this.useDedicatedTenancy = useDedicatedTenancy;
         this.connectBySSHProcess = connectBySSHProcess;
         this.maxTotalUses = maxTotalUses;
+        this.nodeProperties = new DescribableList<NodeProperty<?>,NodePropertyDescriptor>(Saveable.NOOP, nodeProperties);
```

Comment:
- the type parameters on the constructor are redundant, using <> should be sufficient.
- the nodeProperties argument can be null (e.g. in unit tests), this causes a NullPointerException in the constructor of the DescribableList. Wrapping the argument with Util.fixNull(...) should prevent this.